pkg/compiler/codegen.go

```diff
@@ -1635,10 +1635,26 @@ func (c *codegen) convertBuiltin(expr *ast.CallExpr) {
 		} else {
 			emit.Instruction(c.prog.BinWriter, opcode.JMPIFNOT, []byte{2 + 2})
 			emit.Opcodes(c.prog.BinWriter, opcode.DROP, opcode.NEWARRAY0)
-			for _, e := range expr.Args[1:] {
-				emit.Opcodes(c.prog.BinWriter, opcode.DUP)
-				ast.Walk(c, e)
-				emit.Opcodes(c.prog.BinWriter, opcode.APPEND)
+			if expr.Ellipsis.IsValid() {
+				ast.Walk(c, expr.Args[1])                                       // append(x, y...) -> x, y on stack
+				emit.Opcodes(c.prog.BinWriter, opcode.DUP, opcode.REVERSEITEMS) // POPITEM pops last element
```

Comment:
But this will change `y`.

Reply:
Fixed, inserted conversion to `Struct`, added test.

Reply:
`REVERSEITEMS` is quite costly and now we also `CONVERT`, maybe it's easier with `PICKITEM`?

Reply:
Fixed.